### PR TITLE
VEN-534 | Large vessels

### PR DESCRIPTION
**Pull request by a-rmz**
## Description :sparkles:
- Add a `intended_use` text field to `Boat`
- Remove the `boat_is_inspected` and `boat_is_insured` from the `Boat` model
- Add a `BoatCertificate` model
- Add utils to `utils/relay.py` to handle global IDs
- Move `remove_file` to `utils/files.py` to reuse it

### Closes :no_good_woman:
**[VEN-534](https://helsinkisolutionoffice.atlassian.net/browse/VEN-534):** Update Large vessel specific fields on Boat model

### Automated tests :gear:️
```shell
$ pytest customers/tests/test_customer_models.py
$ pytest customers/tests/test_customer_queries.py
$ pytest customers/tests/test_customer_mutations.py
```

### Manual tests 👷 
**Query:**
```graphql
query BOATS {
  berthProfile(id: "") {
    id
    boats {
      edges {
        node { 
          id
          name
          certificates {
            id
            certificateType
            checkedAt
            checkedBy  
            validUntil
            file
          }
        }
      }
    }
  }
}
```

**Create:**
```graphql
mutation CREATE_CERTIFICATE($input:CreateBoatCertificateMutationInput!) {
  createBoatCertificate(input: $input) {
    boatCertificate {
      id
      certificateType
      checkedAt
      checkedBy
      validUntil
    }
  }
}
```

```json
{
  "input": {
    "boatId": "",
    "certificateType": "INSURANCE"
  }
}
```

**Update:**
```graphql
mutation UPDATE_CERTIFICATE($input:UpdateBoatCertificateMutationInput!) {
  updateBoatCertificate(input: $input) {
    boatCertificate {
      id
      certificateType
      checkedAt
      checkedBy
      validUntil
    }
  }
}
```

```json
{
  "input": {
    "id": "",
    "checkedBy": null
  }
}
```

**Delete:**
```graphql
mutation DELETE_CERTIFICATE($input:DeleteBoatCertificateMutationInput!) {
  deleteBoatCertificate(input: $input) {
    __typename
  }
}
```

```json
{
  "input": {
    "id": ""
  }
}
```

## Screenshots :camera_flash:
image

### CODE REVIEW

File: customers/enums.py [hunk 1]
```diff
@@ -24,3 +24,12 @@ class Labels:
...
         OTHER = _("Other")
+
+
+class BoatCertificateType(Enum):
+    INSPECTION = "inspection"
+    INSURANCE = "insurance"
+
+    class Labels:
+        INSPECTION = _("inspection")
+        INSURANCE = _("insurance")
```
Comment (igordavydsson) [inline customers/enums.py:35]: these usually start with a capital letter

File: customers/schema.py [hunk 2]
```diff
@@ -1,35 +1,63 @@
...
+class BoatCertificateNode(DjangoObjectType):
+    certificate_type = BoatCertificateTypeEnum(required=True)
+
+    class Meta:
+        model = BoatCertificate
+        interfaces = (relay.Node,)
+
+    def resolve_file(self, info, **kwargs):
+        if self.file:
+            return info.context.build_absolute_uri(self.file.url)
```
Comment (igordavydsson) [inline customers/schema.py:45]: nitpick, but maybe oneline it:
```python
return info.context.build_absolute_uri(self.file.url) if self.file else None
```

File: customers/schema.py [hunk 3]
```diff
@@ -144,6 +172,98 @@ def get_node(cls, info, id):
...
+        boat_id = graphene.ID(required=True)
+        certificate_type = BoatCertificateTypeEnum(required=True)
+
+    boat_certificate = graphene.Field(BoatCertificateNode)
+
+    @classmethod
+    @add_permission_required(BoatCertificate)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **input):
+        boat = get_node_from_global_id(info, input.pop("boat_id"), only_type=BoatNode)
```
Comment (igordavydsson) [inline customers/schema.py:194]: should this be `nullable=False`?

Comment (igordavydsson) [reply customers/schema.py:194]: ah, it will raise an error later on, ok, no need then 👍

File: customers/schema.py (hunk 2 above)
Comment (a-rmz) [reply customers/schema.py:45]: I was following the example from [resources/schema.py](https://github.com/City-of-Helsinki/berth-reservations/blob/develop/resources/schema.py#L299)
```python
def resolve_image_file(self, info, **kwargs):
    if self.image_file:
        return info.context.build_absolute_uri(self.image_file.url)
    else:
        return None
```

but definitely nicer as a one-liner. 🔥

Comment (igordavydsson) [reply customers/schema.py:45]: we could also change it to a oneliner there 😬

Comment (igordavydsson) [reply customers/schema.py:45]: hm, or was it so that black formats it to several lines anyway 🤔

File: customers/schema.py (hunk 3 above)
Comment (a-rmz) [reply customers/schema.py:194]: Ahh, but it actually should be `nullable=False`, good catch.

File: customers/schema.py (hunk 2 above)
Comment (a-rmz) [reply customers/schema.py]: Works just fine as one-liner. 👌

File: customers/tests/test_customers_mutations.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+
+    variables = {
+        "boatId": boat_id,
+        "file": SimpleUploadedFile(
+            name=file_name, content=None, content_type="application/pdf"
+        ),
+        "certificateType": random.choice(list(BoatCertificateType)).name,
+        "validUntil": check_date,
+        "checkedAt": check_date,
+        "checkedBy": "John Wick",
```
Comment (igordavydsson) [inline customers/tests/test_customers_mutations.py:55]: 👀 😮

Comment (a-rmz) [reply customers/tests/test_customers_mutations.py:55]: Better don't mess with it...

**Pull request merged**